### PR TITLE
AF-45: Added OHHTTPStubs framework so we can test with mock data

### PR DESCRIPTION
**Pull request by mikew-personal**
#### Ticket

[AF-45](https://vimean.atlassian.net/browse/AF-45)

#### Pull Request Checklist

- [x] Resolved any merge conflicts
- [x] No build errors or warnings are introduced
- [x] New files are written in Swift
- [x] New classes contain license headers
- [x] New classes have Documentation
- [x] New public methods have Documentation

#### Issue Summary

- While looking for methods of testing with mock data I came across the OHHTTPStubs framework which was mentioned on the AFNetworking GitHub page as a way of testing network code.  The framework allows us to easily test a request using the VimeoClient object by setting up mocks which return JSON data that we can easily record and store in text files using either curl or Postman.  The framework works by hooking into Cocoa's URL loading system and intercepting requests before they're sent out over the network.  Instead, the framework can return the mock data that you define in your test code to the URL loading system so all of the network code built on top of it (including any AFNetworking code) will treat the data as if it was returned by the endpoint.  The mocking methods allow us to intercept network requests in a number of different ways, including: matching host names, matching HTTP methods (GET, POST, PATCH, etc...), matching endpoint paths (ie. /programmed/cinema), and matching query params.

#### Implementation Summary

- Podfile - I added the OHHTTPStubs dependency to both the VimeoNetworkingExample-iOS and VimeoNetworkingExample-tvOS test targets.  The dependency was not added to the VimeoNetworking target, so it is only accessible in our unit tests.

- Request+ProgrammedContent.swift - New unit tests which exercise the Request+ProgrammedContent.swift code.  These tests demonstrate how to use OHHTTPStubs to mock both a successful request and a failure request.

#### Reviewer Tips

- Most of the changes in this PR are from adding the new dependency. 

#### How to Test

- Run the unit tests

### CODE REVIEW

**Comment by cameo-github** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoNetworking/blob/b4e24fd5e114cb18ad0f5e25bffdc035116fd401/VimeoNetworking/Sources/VimeoClient.swift#L330">VimeoNetworking/Sources/VimeoClient.swift#L330</a></strong>: cast from ‘Response&lt;VIMNullResponse&gt;’ to unrelated type ‘Response&lt;ModelType&gt;’ always fails  <br /><code>
                let response = Response(model: nullResponseObject, json: [:]) as! Response&lt;ModelType&gt;
</code></td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 50 tests, with 0 failures (0 unexpected) in 0.818 (0.856) seconds</td>
    </tr>
  </tbody>
</table>

See build details on [CircleCI](https://circleci.com/gh/vimeo/VimeoNetworking/407)

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: VimeoNetworkingExample-iOS/VimeoNetworkingCommonTests/Request+ProgrammedContent.swift [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        }
+    }
+    
+    func test_CinemaRequest_onFailure_returnsError()
+    {
+        let request = CinemaContentRequest.getCinemaContentRequest()
+        
+        stub(condition: isPath("/programmed/cinema")) { _ in
+            
+            let fakeErrorJSON = ["link": NSNull(),
```
Comment (supperspidey) [inline VimeoNetworkingExample-iOS/VimeoNetworkingCommonTests/Request+ProgrammedContent.swift]: It'll be nice if we have a file whose content is this JSON blob, then just like in the first test case, we can just read off from that file. In addition, we can refactor out the code that reads from JSON files into a function.

Comment (mikew-personal) [inline VimeoNetworkingExample-iOS/VimeoNetworkingCommonTests/Request+ProgrammedContent.swift]: This was more of a demo to show a different method for creating the stubbed data.  The info in the JSON file in the previous test was captured from the server using Postman, so it was easy to put into a file.  For this test, I couldn't capture a live failure from the server so this shows how you might enter in error information using a regular Dictionary if you know what keys the code you're testing is looking for.  As for refactoring the file reading code, I'd have to see more examples of it being used before we can see how it should be wrapped.  The OHHttpStubs framework has a convenient method for reading from a file, so it wouldn't make sense to just wrap it with another function unless there was something like validation code that we kept re-writing before using the framework's file reading method.

**Comment by mikew-personal** [comment]
@supperspidey does my comment address your issue?  If so, can you approve this PR so I can merge it back?

**Comment by supperspidey** [comment]
@mikew-personal yes! Does anyone else have feedback? If not then this PR is ready to merge.

**Pull request merged**